plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,1089 @@
+
+        Args:
+            self (object): The instance of the class.
+            module (object): The Ansible module object.
+            import_license_url (str): The URL for importing the license.
+            resource_id (str): The ID of the resource.
+
+        Returns:
+            object: The import status of the license.
+        """
+        payload = {}
+        payload["ShareType"] = "CIFS"
+        payload["LicenseName"] = module.params.get('share_parameters').get('file_name')
+        payload["FQDD"] = resource_id
+        payload["ImportOptions"] = "Force"
+        if module.params.get('share_parameters').get('workgroup'):
+            payload["Workgroup"] = module.params.get('share_parameters').get('workgroup')
+        share_details = License.get_share_details(self, module)
+        payload.update(share_details)
+        import_status = self.idrac.invoke_request(import_license_url, "POST", data=payload)
+        return import_status
+
+    def __import_license_nfs(self, module, import_license_url, resource_id):
+        """
+        Import a license from an NFS share.
+
+        Args:
+            module (object): The Ansible module object.
+            import_license_url (str): The URL for importing the license.
+            resource_id (str): The ID of the resource.
+
+        Returns:
+            dict: The import status of the license.
+        """
+        payload = {}
+        payload["ShareType"] = "NFS"
+        payload["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+        payload["ShareName"] = module.params.get('share_parameters').get('share_name')
+        payload["LicenseName"] = module.params.get('share_parameters').get('file_name')
+        payload["FQDD"] = resource_id
+        payload["ImportOptions"] = "Force"
+        import_status = self.idrac.invoke_request(import_license_url, "POST", data=payload)
+        return import_status
+
+    def __check_file_extension(self, module):
+        """
+        Check if the file extension of the given file name is valid.
+
+        :param module: The Ansible module object.
+        :type module: AnsibleModule
+
+        :return: None
+        """
+        share_type = module.params.get('share_parameters').get('share_type')
+        file_name = module.params.get('share_parameters').get('file_name')
+        valid_extensions = {".txt", ".xml"} if share_type == "local" else {".xml"}
+        file_extension = any(file_name.lower().endswith(ext) for ext in valid_extensions)
+        if not file_extension:
+            module.exit_json(msg=INVALID_FILE_MSG, failed=True)
+
+    def __get_import_license_url(self, module):
+        """
+        Get the import license URL.
+
+        :param module: The module object.
+        :type module: object
+        :return: The import license URL.
+        :rtype: str
+        """
+        uri, error_msg = validate_and_get_first_resource_id_uri(
+            self.module, self.idrac, MANAGERS_URI)
```

Comment:
Here i see self.module used and in some places i see

module used can we keep it consistent to use self.module and remove passing the module param internally.

Same applies for all the methods.

Reply:
Done